OTAnalytics/plugin_parser/streaming_parser.py

```diff
@@ -0,0 +1,349 @@
+            Defaults to [].
+        registered_videos_metadata (list[VideosMetadata], optional):
+            VideosMetadata objects to be updated with each parsed files metadata.
+            Defaults to [].
+        progressbar (ProgressbarBuilder, optional):
+            a progressbar builder to show progress of processed files.
+            Defaults to TqdmBuilder().
+        track_dataset_factory (TrackDataSetFactory, optional):
+            a factory to create a new track dataset from a list of Tracks.
+            Defaults to PandasTrackDataset.from_list(tracks,
+            PygeosTrackGeometryDataset.from_track_dataset, PandasByMaxConfidence()).
+        chunk_size (int, optional): defines the number of tracks to be collected,
+            before yielding a TrackDataset containing at most that many Tracks.
+            Defaults to 1.
+    """
+
+    def __init__(
+        self,
+        detection_parser: StreamDetectionParser,
+        format_fixer: OttrkFormatFixer = OttrkFormatFixer(),
+        registered_tracks_metadata: list[TracksMetadata] = [],
+        registered_videos_metadata: list[VideosMetadata] = [],
+        progressbar: ProgressbarBuilder = TqdmBuilder(),
+        track_dataset_factory: TrackDatasetFactory = default_track_dataset_factory,
+        chunk_size: int = 5,
+    ) -> None:
+        self._detection_parser = detection_parser
+        self._tracks_dict: dict[TrackId, list[Detection]] = {}
+        self._format_fixer = format_fixer
+        self._registered_tracks_metadata: set[TracksMetadata] = set(
+            registered_tracks_metadata
+        )
+        self._registered_videos_metadata: set[VideosMetadata] = set(
+            registered_videos_metadata
+        )
+        self._progressbar = progressbar
+        self._track_dataset_factory = track_dataset_factory
+        self._chunk_size = chunk_size
+
+    def register_tracks_metadata(self, tracks_metadata: TracksMetadata) -> None:
+        """Register TracksMetadata to be updated when a new ottrk file is parsed."""
+        self._registered_tracks_metadata.add(tracks_metadata)
+
+    def register_videos_metadata(self, videos_metadata: VideosMetadata) -> None:
+        """Register VideosMetadata to be updated when a new ottrk file is parsed."""
+        self._registered_videos_metadata.add(videos_metadata)
+
+    def _update_registered_metadata_collections(
+        self,
+        new_detection_metadata: DetectionMetadata,
+        new_video_metadata: VideoMetadata,
+    ) -> None:
+
+        for tracks_metadata in self._registered_tracks_metadata:
+            tracks_metadata.update_detection_classes(
+                new_detection_metadata.detection_classes
+            )
+
+        for videos_metadata in self._registered_videos_metadata:
+            videos_metadata.update(new_video_metadata)
+
+    def parse(self, files: set[Path]) -> Iterator[TrackDataset]:
+        iterator = iter(self._parse_tracks(files))
+        while True:
```

Comment:
When does the loop end?

Reply:
once the iterator providing the parsed tracks from the files runs out of tracks, the resulting chunk will be an empty list and the break condition inside the loop will trigger

Reply:
Should we use an empty result here or raise a StopIteration?

Reply:
Using break and return should both have the same behavior: the generator function ends and a StopIteration is raised. 
But using a return statment, makes the intent / meaning clearer. Changed in 124eeed